### DISCUSSION TITLE
Battery voltage reported value

### DISCUSSION THREAD

**Discussion by lvenditti** [flobz/psa_car_controller#354] [Ideas]
Looking at battery voltage reported by ODBII on car scanner (or similar) APP, it is a bit higher than value reported by HA using rest sensor.
My unsteranding is that the sensor takes read value x 4 instead should be read value x 5.
In this way reported value on Home Assistant and on ODBII dongle are pretty close

Ciao.

**Comment by gerar19** [comment]
@flobz 
The problem is not that, but that the voltage refers to the 12V battery, not to the traction one.
I know because I'm having phantom consumption problems on my e208 and the voltage values ​​have dropped to 0.
Taking as a reference the raw value obtained from the voltage, which is a % from 0 to 100, the result matches 100% with the value obtained by ODB pot of the 12V battery.

So you would have to change the yaml sensor to this:
    mycar_battery_voltage:
      friendly_name: "MyCar Battery Voltage"
      unit_of_measurement: "V"
      value_template: '{{ states.sensor.mycar.attributes["battery"]["voltage"] * 13.2 / 100 | round(2) }}'
      device_class: voltage
